tools/common/lib.go

```diff
@@ -1079,3 +1080,148 @@ func getIfSetString(c *cli.Context, p string, setCount *int) (r *string) {
 	}
 	return
 }
+
+type storeClientCache struct {
+	mClient mcli.Client
+	cache   map[string]*storehost.StoreClientImpl
+}
+
+func newStoreClientCache(mClient mcli.Client) *storeClientCache {
+	return &storeClientCache{
+		mClient: mClient,
+		cache:   make(map[string]*storehost.StoreClientImpl),
+	}
+}
+
+func (t *storeClientCache) get(storeUUID string) (*storehost.StoreClientImpl, error) {
+
+	client, ok := t.cache[storeUUID]
+
+	if !ok {
+		hostAddr, err := t.mClient.UUIDToHostAddr(storeUUID)
+		if err != nil {
+			hostAddr = storeUUID + UnknownUUID
```

Comment:
i am not 100% sure .. but that seems to be the pattern we have used throughout the code. my sense is it is to ensure we print it out the error with the UUID, etc.